src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -158,6 +161,16 @@ static MvcCoreLoggerExtensions()
                 2,
                 "Executed action {ActionName} in {ElapsedMilliseconds}ms");
 
+            _pageExecuting = LoggerMessage.Define<string, string>(
+                LogLevel.Information,
+                3,
+                "Route matched with {RouteData}. Executing page {ActionName}");
```

Comment:
`ActionName` -> `PageName` here and below. I realize you're passing `ActionDescriptor.DisplayName` into these but should be specific.